From 63e0e4c3ea85be127ad36fc0e635ac4f2063fea4 Mon Sep 17 00:00:00 2001
From: Emilio Pozuelo Monfort <pochu27@gmail.com>
Date: Tue, 25 Mar 2014 13:44:34 +0100
Subject: [PATCH] wayland: Add missing CLUTTER_AVAILABLE annotations

Signed-off-by: Emilio Pozuelo Monfort <pochu27@gmail.com>
---
 clutter/wayland/clutter-wayland-surface.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/clutter/wayland/clutter-wayland-surface.h b/clutter/wayland/clutter-wayland-surface.h
index da051e4..f674af8 100644
--- a/clutter/wayland/clutter-wayland-surface.h
+++ b/clutter/wayland/clutter-wayland-surface.h
@@ -88,21 +88,28 @@ struct _ClutterWaylandSurfaceClass
   gpointer _padding_dummy[8];
 };
 
+CLUTTER_AVAILABLE_IN_1_10
 GType clutter_wayland_surface_get_type (void) G_GNUC_CONST;
 
+CLUTTER_AVAILABLE_IN_1_8
 ClutterActor *clutter_wayland_surface_new               (struct wl_surface *surface);
+CLUTTER_AVAILABLE_IN_1_10
 void          clutter_wayland_surface_set_surface       (ClutterWaylandSurface *self,
                                                          struct wl_surface *surface);
+CLUTTER_AVAILABLE_IN_1_10
 struct wl_surface *clutter_wayland_surface_get_surface  (ClutterWaylandSurface *self);
+CLUTTER_AVAILABLE_IN_1_8
 gboolean      clutter_wayland_surface_attach_buffer     (ClutterWaylandSurface *self,
                                                          struct wl_resource *buffer,
                                                          GError **error);
+CLUTTER_AVAILABLE_IN_1_8
 void          clutter_wayland_surface_damage_buffer     (ClutterWaylandSurface *self,
                                                          struct wl_resource *buffer,
                                                          gint32 x,
                                                          gint32 y,
                                                          gint32 width,
                                                          gint32 height);
+CLUTTER_AVAILABLE_IN_1_10
 CoglTexture  *clutter_wayland_surface_get_cogl_texture  (ClutterWaylandSurface *self);
 
 G_END_DECLS
-- 
1.9.0

