Description: Build without the cookbook directory
Author: Jeremy Bicha <jbicha@linux.com>
Forwarded: no

--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -4,11 +4,7 @@ if BUILD_GTK_DOC
 SUBDIRS += reference
 endif
 
-if ENABLE_DOCS
-SUBDIRS += cookbook
-endif
-
-DIST_SUBDIRS = reference cookbook
+DIST_SUBDIRS = reference
 
 EXTRA_DIST = \
 	CODING_STYLE.txt \
--- a/configure.ac
+++ b/configure.ac
@@ -1207,9 +1207,6 @@ AC_CONFIG_FILES([
 
 	doc/Makefile
 	doc/reference/Makefile
-	doc/cookbook/Makefile
-	doc/cookbook/clutter-cookbook.xml
-	doc/cookbook/examples/Makefile
 
 	po/Makefile.in
 ])
--- a/Makefile.am
+++ b/Makefile.am
@@ -30,7 +30,6 @@ meson_files = \
 	clutter/meson.build \
 	clutter/cex100/meson.build \
 	doc/meson.build \
-	doc/cookbook/meson.build \
 	doc/reference/meson.build \
 	doc/reference/xml/gtkdocentities.ent.in \
 	doc/reference/xml/meson.build \
