From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 10 Mar 2023 08:04:35 +0100
Subject: configure.ac: Use proper clutter micro version

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b8f6f35..55b147f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,7 +10,7 @@
 # - increase clutter_interface_version to the next odd number
 m4_define([clutter_major_version], [1])
 m4_define([clutter_minor_version], [26])
-m4_define([clutter_micro_version], [5])
+m4_define([clutter_micro_version], [4])
 
 # • for stable releases: increase the interface age by 1 for each release
 # • for development releases: keep clutter_interface_age to 0
